homeassistant/components/generic/camera.py

```diff
@@ -62,7 +60,7 @@
             cv.small_float, cv.positive_int
         ),
         vol.Optional(CONF_VERIFY_SSL, default=True): cv.boolean,
-        vol.Optional(CONF_RTSP_TRANSPORT): vol.In(RTSP_TRANSPORTS.keys()),
+        vol.Optional(CONF_RTSP_TRANSPORT): vol.In(RTSP_TRANS_PROTOCOLS),
```

Comment:
Yeah I noticed this too. Basically the same information was in generic and onvif, but generic also stored the labels in a dict. We could still use a dict like the one used in generic, and that could be used to print stuff a little nicer. At the same time it also seems a bit roundabout since it's just for config_flows, but stream itself doesn't have a config_flow.

Reply:
I think the dict with the friendly names should be retained.  They can be mapped back to the raw strings when the config flow processes the user input, and this can be contained within generic.

I'm hoping to make some more of this automatic to eliminate the need for the user option.  But if that is not successful we may need to put some helper text in there. e.g. 
"HTTP (usual option for most setups)"
"UDP (sometimes needed for xyz)"
....

Reply:
Sure